src/ubloxGPS.cpp

```diff
@@ -862,6 +855,30 @@ bool ubloxGPS::setBaudrate(ubx_baudrate_t baudrate)
     return (bool) (waitForAck(UBX_CLASS_CFG, UBX_CFG_PRT) != NULL);
 }
 
+/**
+ * Set configuration for ubloxGPS SPI port
+ * 
+ */
+bool ubloxGPS::setSpiMode(int txReady, int threshold, int spiMode) {
```

Comment:
Makes sense to break this out. My understanding is that this cfg setting can be persisted to internal flash and preserved across resets; however it sounds like if the D_SEL hw line is not well coupled to ground, the ublox default behavior is to switch back to uart mode. Consider capturing this tribal knowledge somewhere such as code comments.

Reply:
I agree.  It would have to be programmed at least once at manufacturing or checked at startup.  One concern is that the customer can fork this repo and make their own changes that deviate from the settings here.